### PR TITLE
[MIG] sale_order_price_recalculation: Migration to 10.0

### PR DESCRIPTION
**Pull request by chienandalu**
Recalculation of sales order lines prices on demand
===================================================

This module add 2 buttons on sale orders (below sale order lines) that:

* Recalculates the prices of the order lines that contain a product in them.
* Reset product descriptions from current product information.

cc @Tecnativa

### CODE REVIEW

File: sale_order_price_recalculation/models/sale_order.py [hunk 1]
```diff
@@ -17,11 +18,7 @@ def recalculate_prices(self):
...
         return True
 
-    @api.multi
     def recalculate_names(self):
         for line in self.mapped('order_line'):
-            name = line.product_id.name_get()[0][1]
-            if line.product_id.description_sale:
-                name += '\n' + line.product_id.description_sale
-            line.name = name
+            line.product_id_change()
```
Comment (pedrobaeza) [inline sale_order_price_recalculation/models/sale_order.py]: Isn't this going to change also prices?

**Comment by chienandalu** [comment]
@pedrobaeza Indeed it was wrong. I wanted to make that function universally connectable (for example for the case of sale_variant_configurator).

File: sale_order_price_recalculation/models/sale_order.py [hunk 2]
```diff
@@ -1,42 +1,36 @@
...
-                qty_uos=line.product_uos_qty, uos=line.product_uos.id,
-                name=line.name, partner_id=order.partner_id.id, lang=False,
-                update_tax=True, date_order=order.date_order, packaging=False,
-                fiscal_position=order.fiscal_position.id, flag=price)
-            if price:
-                line.write(res['value'])
-            else:
-                if 'name' in res['value']:
-                    line.write({'name': res['value']['name']})
+            if line.product_id:
```
Comment (pedrobaeza) [inline sale_order_price_recalculation/models/sale_order.py]: Try this to see if it works:
```
for line in self.mapped('order_line').filtered('product_id'):
    line2 = line.with_context()  # we make this to isolate changed values
    line2.product_uom_change()
    line.name = line2.name
```

**Comment by chienandalu** [comment]
@pedrobaeza, that's a great approach that is neutral with the cases I exposed :slightly_smiling_face: The onchange to use has to be `product_id_change`, anyway. And `line2 = line.with_context()` didn't isolate enough the original object in this case. Reused the trick you gave me before :smile:

File: sale_order_price_recalculation/models/sale_order.py [hunk 3]
```diff
@@ -1,42 +1,42 @@
...
 
 
 class SaleOrder(models.Model):
     _inherit = 'sale.order'
 
     @api.multi
     def recalculate_prices(self):
-        return self.reset_lines(price=True)
+        for line in self.mapped('order_line'):
+            dict = {
```
Comment (pedrobaeza) [inline sale_order_price_recalculation/models/sale_order.py]: Please use the read technique we have seen in other places

Comment (chienandalu) [reply sale_order_price_recalculation/models/sale_order.py]: I'm no sure about what you mean here @pedrobaeza :confused:

Comment (pedrobaeza) [reply sale_order_price_recalculation/models/sale_order.py]: `dict = line._convert_to_write(line._convert_to_read(line.read()[0]))`

**Comment by chienandalu** [comment]
@pedrobaeza Changes done

**Review by pedrobaeza — APPROVED** [review]
@luismontalba please review

**Review by luismontalba — APPROVED** [review]
Tested on runbot

**Comment by pedrobaeza** [comment]
@chienandalu please squash your commits for merging this.

**Comment by pedrobaeza** [comment]
Please squash for merging

**Comment by chienandalu** [comment]
@pedrobaeza

**Comment by chienandalu** [comment]
@pedrobaeza done

**Pull request merged**